safe-global/safe-wallet-web
# Feat: display fiat balances for ERC20 tokens in counterfactual safes

Blocked: https://5afe.slack.com/archives/C05HTQ67JAJ/p1718902866327009

## What it solves

Resolves #3806

## How this PR fixes it
1. In the sidebar/login page safe list
- For undeployed safes on supported chains, use `/chains/{chainId}/safes/{safeAddress}/balances/{fiatCode}` to get the total fiat balance.
- For deployed safes, get the fiat total from the safe overviews, as before.
- For undelpoyed safes on unsupported chains, show nothing. (remove loading skeleton)
2. On the assets page
- For undeployed safes on supported chains, get ERC20 asset balances from `/chains/{chainId}/safes/{safeAddress}/balances/{fiatCode}`
- For undelpoyed safes on unsupported chains, only fetch the native token balance, as before.

## How to test it
- Create a counterfactual safe on one of the supported chains (https://5afe.slack.com/archives/C04MJJM9S3U/p1717772188776199?thread_ts=1717673628.428579&cid=C04MJJM9S3U)
- fund the safe with ERC20 tokens.
- Check that the fiat conversion of the ERC20 balances show in the assets page [1]
- Check that the total fiat balance shows correctly in the safe list in the sidebar and login page. [2]

## Screenshots
[1]
![image](https://github.com/safe-global/safe-wallet-web/assets/17801424/ea06aeee-92ad-4c8c-8e8f-b49e25ea6ca0)
[2]
![image](https://github.com/safe-global/safe-wallet-web/assets/17801424/d6b5966e-1647-4b62-9ba1-07e3aef05df5)


## Checklist
